e2e/cypress/integration/messaging/mention_autocomplete_overlap_spec.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18667-At-mention user autocomplete does not overlap with channel header when drafting a long message containing a file attachment', () => {
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Upload file
+        cy.fileUpload('#fileUploadInput');
+
+        // # Create and then type message to use
+        let message = 'h{shift}';
+        cy.get('#post_textbox').clear();
+        for (let i = 0; i < 12; i++) {
+            message += '{enter}h';
+        }
+        cy.get('#post_textbox').type(message);
+
+        // # Put cursor on top and add the mention
+        let returnMessage = '';
+        for (let i = 0; i < 12; i++) {
+            returnMessage += '{uparrow}';
```

Comment:
I don't think we need this. We want the add the `@mention` at the end of the string of `h`'s rather than the beginning. It would be better to do another `shift + enter` and then put the `@mention`.

Reply:
Here I was trying to force the test to its limits. I understand that the mention on top of the message will cause more problems than the mention at the bottom. Nevertheless, if the test makes more sense at the bottom, I have no problem changing that.

Reply:
This is due to the specs of the test. Although not mentioned in the ticket (wasn't clarified) but we would prefer to have the `@` at the end of the text rather than the beginning.